crainets/models/efficientnet3d/mbconvblock3d.py

```diff
@@ -0,0 +1,167 @@
+            self.norm0 = self._norm(norm=norm, output=self.mid_channels, bias=True)
+
+        # Depthwise convolution phase
+
+        self.depthwise_conv = Conv3d(
+            in_channels=self.mid_channels,
+            out_channels=self.mid_channels,
+            groups=self.mid_channels,  # groups makes it depthwise
+            kernel_size=kernel_size,
+            stride=stride,
+            bias=False,
+            )
+
+        self.norm1 = self._norm(norm=norm, output=self.mid_channels, bias=True)
+
+        # Squeeze and Excitation layer, if desired
+        if self.has_se:
+            # ratio such that the number of squeezed channels are ratio(0.25)*in_channels from efficientnet architecture
+            num_squeezed_channels = max(1, se_ratio*self.in_channels/self.mid_channels)
+            self.se_reduce = Conv3d(in_channels=self.mid_channels, out_channels=num_squeezed_channels, kernel_size=1)
+            self.se_expand = Conv3d(in_channels=num_squeezed_channels, out_channels=self.mid_channels, kernel_size=1)
+
+
+        # Pointwise convolution phase
+        self.project_conv = Conv3d(in_channels=self.mid_channels,
+                                   out_channels=out_channels,
+                                   kernel_size=1,
+                                   bias=False,
+                                   )
+        self.norm2 = self._norm(norm=norm, output=out_channels, bias=True)
+        self.silu = torch.nn.SiLU()
+
+    def forward(self, inputs, drop_connect_rate=None):
+        """MBConvBlock's forward function.
+        Args:
+            inputs (tensor): Input tensor.
+            drop_connect_rate (bool): Drop connect rate (float, between 0 and 1).
+        Returns:
+            Output of this block after processing.
+        """
+
+        # Expansion and Depthwise Convolution
+        x = inputs
+        if self.expand_ratio != 1:
+            x = self.expand_conv(inputs)
+            x = self.norm0(x)
+            x = self.silu(x)
+
+        x = self.depthwise_conv(x)
+        x = self.norm1(x)
+        x = self.silu(x)
+
+        # Squeeze and Excitation
+        if self.has_se:
+            x_squeezed = F.adaptive_avg_pool3d(x, 1)
+            x_squeezed = self.se_reduce(x_squeezed)
+            x_squeezed = self.silu(x_squeezed)
+            x_squeezed = self.se_expand(x_squeezed)
+            x = torch.sigmoid(x_squeezed) * x
+
+        # Pointwise Convolution
+        x = self.project_conv(x)
+        x = self.norm2(x)
+
+        # Skip connection and drop connect
+        if self.id_skip and self.stride == 1 and self.in_channels == self.out_channels:
+            # The combination of skip connection and drop connect brings about stochastic depth.
+            if drop_connect_rate:
+                x = drop_connect(x, p=drop_connect_rate, training=self.training)
+            x = x + inputs  # skip connection
+
+        return x
+
+
+    def _norm(self, norm: str, output: int, bias: bool):
```

Comment:
This function could be made nicer with dicts and setattr. Repeated code is a pain to maintain